tests/dialects/test_mpi_lowering.py

```diff
@@ -0,0 +1,269 @@
+from xdsl.dialects import mpi, func, llvm, builtin
+from xdsl.ir import Operation, Attribute, OpResult, SSAValue
+from xdsl.irdl import irdl_op_definition, VarOpResult
+from xdsl.transforms import mpi_to_llvm
+
+info = mpi_to_llvm.MpiLibraryInfo()
+
+
+def extract_func_call(ops: list[Operation],
+                      name: str = 'MPI_') -> func.Call | None:
+    for op in ops:
+        if isinstance(op,
+                      func.Call) and op.callee.string_value().startswith(name):
+            return op
+
+
+def check_emitted_function_signature(ops: list[Operation],
+                                     name: str,
+                                     types: tuple[type[Attribute] | None, ...],
+                                     ignore_list: tuple[SSAValue] = tuple()):
+    call = extract_func_call(ops, name)
+    assert call is not None, f"Missing func.Call op to {name} in output!"
+    assert len(call.arguments) == len(types)
+    for i, type_ in enumerate(types):
```

Comment:
```suggestion
    for i, typ in enumerate(types):
```
Trailing underscore looks a little funky